doc/style_guide.rst

```diff
@@ -311,3 +346,95 @@ Miscellaneous
 
 - Likewise with a setter API, replacing `foo` with `foo=` and `getFoo` with `setFoo`
   in the above text.
+
+Best practices
```

Comment:
"best practices" applies to coding style, so it does not make sense to keep these things separate in the `contribution.rst`

Reply:
Again, excerpt would help, so people don't have to bounce around chasing links, page loads and UI interruptions break learning flow.

Reply:
I don't understand what exactly do I need to excerpt and where to put it. Also don't understand second part of the comment - there are almost no links in this part except for `rationale: https://forum.nim-lang.org/t/4089` one. Did you mean this one? Or avoid linking things and instead excerpt them?

Reply:
While the whole compiler and bootstrap process is very fragile, the extra coding style for the committee is to avoid macros.

Doesn't need to be in this file, just realized there are some situational specifics.

Reply:
best practices to add, but can be a separate PR if you want:

- avoid the boolean trap (use enums)
- avoid primitive or very general return types unless truly operating on them, prefer named types especially if many general operations so not apply or uncertain about the API

Reply:
> I don't understand what exactly do I need to excerpt and where to put it. Also don't understand second part of the comment - there are almost no links in this part except for `rationale: https://forum.nim-lang.org/t/4089` one. Did you mean this one? Or avoid linking things and instead excerpt them in general?

Nothing major, just that being able to excerpt from the style guide into the contribution guide would be nice. So we don't duplicate the same information source, but can reuse it in many places.